NetKet/GroundState/exact_diagonalization.hpp

```diff
@@ -23,52 +23,98 @@
 #include "Hamiltonian/MatrixWrapper/matrix_wrapper.hpp"
 
 namespace netket {
+  
+  namespace detail {
+    template <class matrix_t, class iter_t, class random_t>
+    std::vector<double>eigenvalues_lanczos_run(const matrix_t& matrix,
```

Comment:
A minor point regarding formatting: According to the Google Style Guide [function names](https://google.github.io/styleguide/cppguide.html#Function_Names) should use `MixedCaseLikeThis`, which is used for the rest of NetKet as well.